src/xtgeo/well/_well_oper.py

```diff
@@ -435,6 +435,38 @@ def mask_shoulderbeds(self, inputlogs, targetlogs, nsamples, strict):
     """Mask targetlogs around discrete boundaries."""
     logger.info("Mask shoulderbeds for some logs...")
 
+    useinputs, usetargets, use_numeric = _mask_shoulderbeds_checks(
+        self, inputlogs, targetlogs, nsamples, strict
+    )
+
+    if not useinputs or not usetargets:
+        logger.info("Mask shoulderbeds for some logs... nothing done")
+        return False
+
+    for inlog in useinputs:
+        inseries = self._df[inlog]
+        if use_numeric:
+            bseries = _get_bseries(inseries, nsamples)
+        else:
+            mode, value = list(nsamples.items())[0]
+
+            depth = self._df["Z_TVDSS"]
+            if mode == "md" and self.mdlogname is not None:
+                depth = self._df[self.mdlogname]
+            elif mode == "md" and self.mdlogname is None:
+                raise ValueError("There is no mdlogname attribute present.")
+
+            bseries = _get_bseries_by_distance(depth, inseries, value)
+
+        for target in usetargets:
+            self._df.loc[bseries, target] = np.nan
+
+    logger.info("Mask shoulderbeds for some logs... done")
+    return True
+
+
+def _mask_shoulderbeds_checks(self, inputlogs, targetlogs, nsamples, strict):
+    """Checks/validates input for mask targetlogs around discrete boundaries."""
```

Comment:
document return type, 3-tuple

Reply:
The requirements for docstrings details in private functions are (much) less strict and developers should easily understand by reading code. But I will add a comment.